### PR TITLE
Prepare new version

### PR DESCRIPTION
**Pull request by wdevazelhes**
This PR is to prepare the new release and discuss the process/timeline

### CODE REVIEW

**Comment by bellet** [comment]
@perimosocordiae @terrytangyuan your input is welcome! Our idea is to release soon but maybe there are some points you would like to see investigated a bit further before the release? Potentially related to the last few PRs that we had to merge rather quickly (we were organizing a hands-on session on metric-learn this morning in our research group)

Otherwise we need to carefully write the change log (which will be long)

**Comment by wdevazelhes** [comment]
I started to update the changelog https://github.com/metric-learn/metric-learn/releases, I still need to describe the main API changes of WeaklySupervised algorithms (I think I'll point to the documentation too so that it's simpler), let me know what you think otherwise

**Comment by bellet** [comment]
Nice, I made a few small edits. For weakly supervised algorithms, I think it would be nice to describe at a high level the main changes and their concrete implications (model selection / hyperparameter tuning, scoring, etc). Pointing to the doc for details is fine.

**Comment by bellet** [comment]
Some points that came up during the hands-on session:
- There seemed to be a compatibility problem with the Graphical Lasso import and sklearn v0.20
- Some methods (such as `check_preprocessor`) should probably made private
- NCA with random init change even when setting the `random_state`

**Comment by wdevazelhes** [comment]
- I tried to reproduce the bug with sklearn 0.20.3 but I can't manage to do it. I'll investigate this further
- For the check_preprocessor, I agree, I'll open a PR for that and check that there are no other case like this
- For the random behaviour, indeed, I forgot to add the random_state when initializing the matrix... I'll add it in a PR and add a test that all our estimator that have either a 'init' or a 'prior' are deterministic when setting the random_seed. (I would test that any estimator with a random_seed parameter is deterministic for any value of the others parameters, but unfortunately there's no systematic way to enumerate all the values that the parameters can take for now)

**Comment by wdevazelhes** [comment]
I just opened the 2 PRs, and updated the draft for the release, let me know what you think 
I just need to investigate the problem with sklearn 0.20

**Comment by bellet** [comment]
Looks good, I did minor edits. Let's sort out this sklearn 0.20 thing while we wait for feedback from @perimosocordiae :-)

**Comment by bellet** [comment]
So the sklearn 0.20 thing was a false alarm, it turns out the version for which the problem occurs is 0.19, which is fine since we require 0.20

**Comment by perimosocordiae** [comment]
I think we're just about ready to release! I want to take a look at the current state of the docs one more time before we cut, but code-wise we should be good.

File: doc/conf.py [hunk 1]
```diff
@@ -22,8 +22,8 @@
 project = u'metric-learn'
 copyright = u'2015-2018, CJ Carey and Yuan Tang'
 author = u'CJ Carey and Yuan Tang'
```
Comment (perimosocordiae) [inline doc/conf.py]: I think this line needs some updating as well!

**Comment by bellet** [comment]
> I think we're just about ready to release! I want to take a look at the current state of the docs one more time before we cut, but code-wise we should be good.

Sounds good, feel free to open an issue/PR or let us know if everything is good

File: doc/conf.py (hunk 1 above)
Comment (wdevazelhes) [reply doc/conf.py]: That's right, done, for the order I thought [original developers | my supervisors | me]

**Comment by wdevazelhes** [comment]
@perimosocordiae @terrytangyuan @bellet @nvauquie, I guess we are ready for the release, unless you have any other comment ? For the guidelines I'll follow what we did last time in #71

File: doc/conf.py (hunk 1 above)
Comment (bellet) [reply doc/conf.py]: Please put yourself before Nathalie and I!

**Comment by bellet** [comment]
Otherwise LGTM :-)

File: doc/conf.py (hunk 1 above)
Comment (wdevazelhes) [reply doc/conf.py]: Allright, done ;)

**Comment by terrytangyuan** [comment]
Looks great! Thanks.

**Comment by perimosocordiae** [comment]
Merged.

**Comment by wdevazelhes** [comment]
Great ! I'll lauch the release

**Comment by wdevazelhes** [comment]
Done ! The docs are not yet updated though... Let's wait until tomorrow in case.. (I updated the gh-pages branch with a doc built locally but github does not seem to updated the changes)

**Comment by wdevazelhes** [comment]
I did a wrong process to push the docs in fact, now I remembered and it works ! I guess the release is finalized :tada: :tada:

**Comment by bellet** [comment]
Congrats!
